src/esp/physics/PhysicsManager.cpp

```diff
@@ -189,6 +189,16 @@ void PhysicsManager::stepPhysics(double dt) {
   double targetTime = worldTime_ + dt;
   while (worldTime_ < targetTime) {
     // per fixed-step operations can be added here
+
+    // kinematic velocity control intergration
+    for (auto& object : existingObjects_) {
+      VelocityControl& velControl = object.second->getVelocityControl();
```

Comment:
Usually `std::pair` will only be used in a small area where the definition and application are not too "far away" to each other. This is for readability consideration. Better solution here is to use a `struct`.
Not necessary to change in this PR as it is beyond the scope of this PR.

Reply:
This `second` is actually coming from a std::map iterator for `existingObjects_`, not a std::pair.